stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/feature/messages/MessageListActivity.kt

```diff
@@ -63,10 +63,23 @@ public open class MessageListActivity : AppCompatActivity() {
         private const val EXTRA_CID: String = "extra_cid"
         private const val EXTRA_MESSAGE_ID: String = "extra_message_id"
 
+        /**
+         * Creates an Intent to start the [MessageListActivity] or its subclass.
+         *
+         * @param context The context that will be used in the intent.
+         * @param cid The full channel id, i.e. "messaging:123".
+         * @param messageId The ID of the selected message.
+         * @param activityClass The Activity class that will be used in the intent.
+         */
         @JvmStatic
         @JvmOverloads
-        public fun createIntent(context: Context, cid: String, messageId: String? = null): Intent {
-            return Intent(context, MessageListActivity::class.java).apply {
+        public fun createIntent(
+            context: Context,
+            cid: String,
+            messageId: String? = null,
+            activityClass: Class<out MessageListActivity> = MessageListActivity::class.java
```

Comment:
Not the best API to pass the concrete class, but should be fine taking into account the the Activity is mostly used for a quick start and shouldn't be used in production by the clients.